cellengine/resources/compensation.py

```diff
@@ -59,19 +79,20 @@ def dataframe_as_html(self):
         """Return the compensation matrix dataframe as HTML."""
         return self.dataframe._repr_html_()
 
-    def apply(self, file: FcsFile, inplace: bool = True, **params):
+    def apply(self, file, inplace: bool = False, **kwargs):
```

Comment:
I think you said types only need to be specified in the argument list *or* in the docstring. Do you prefer one vs. the other, and should `bool` be removed, then?

Reply:
I prefer the arg list, since that's what type checkers and IDEs will use

Reply:
Unfortunately, this was motivated by a circular import, which I _though_ was fixed with `from __future__ import annotations`, but that doesn't appear to be working.  This has been fixed (see top of this file), but it's a little ugly. 

Reference here: https://www.python.org/dev/peps/pep-0484/#runtime-or-type-checking

Reply:
It also breaks linting for Python 3.7, but not for 3.8 or 3.9...so I bumped the version in the Actions container.